noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -0,0 +1,557 @@
+
+    // *****************************************************************************
+    // Prepend / append more bytes of data to the ciphertext, before converting back
+    // to fields.
+    // *****************************************************************************
+
+    let mut log_bytes_padding_to_mult_31 =
+        get_arr_of_size__log_bytes_padding__from_PT::<(N * 32) + 64>();
+    /// Safety: this randomness won't be constrained to be random. It's in the
+    /// interest of the executor of this fn to encrypt with random bytes.
+    log_bytes_padding_to_mult_31 = unsafe { get_random_bytes() };
+
+    let mut log_bytes = get_arr_of_size__log_bytes__from_PT::<(N * 32) + 64>();
+
+    assert(
+        log_bytes.len() % 31 == 0,
+        "Unexpected error: log_bytes.len() should be divisible by 31, by construction.",
+    );
+
+    log_bytes[0] = eph_pk_sign_byte;
+    let mut offset = 1;
+    for i in 0..header_ciphertext_bytes.len() {
+        log_bytes[offset + i] = header_ciphertext_bytes[i];
+    }
+    offset += header_ciphertext_bytes.len();
+
+    for i in 0..ciphertext_bytes.len() {
+        log_bytes[offset + i] = ciphertext_bytes[i];
+    }
+    offset += ciphertext_bytes.len();
+
+    for i in 0..log_bytes_padding_to_mult_31.len() {
+        log_bytes[offset + i] = log_bytes_padding_to_mult_31[i];
+    }
+
+    assert(
+        offset + log_bytes_padding_to_mult_31.len() == log_bytes.len(),
+        "Something has gone wrong",
+    );
+
+    // *****************************************************************************
+    // Convert bytes back to fields
+    // *****************************************************************************
+
+    let log_bytes_as_fields = be_bytes_31_to_fields(log_bytes);
+
+    // *****************************************************************************
+    // Prepend / append fields, to create the final log
+    // *****************************************************************************
+
+    // In this strategy, we prepend [tag, eph_pk.x]
+
+    // We assume that the sender wants for the recipient to find the tagged note,
+    // and therefore that they will cooperate and use the correct tag. Usage of a bad
+    // tag will result in the recipient not being able to find the note automatically.
+    let tag = unsafe { get_app_tag_as_sender(sender, recipient) };
+    increment_app_tagging_secret_index_as_sender(sender, recipient);
+
+    let mut final_log: [Field; PRIVATE_LOG_SIZE_IN_FIELDS] = [0; PRIVATE_LOG_SIZE_IN_FIELDS];
+
+    final_log[0] = tag;
+    final_log[1] = eph_pk.x;
+
+    let mut offset = 2;
+    for i in 0..log_bytes_as_fields.len() {
+        final_log[offset + i] = log_bytes_as_fields[i];
+    }
+    offset += log_bytes_as_fields.len();
+
+    for i in offset..PRIVATE_LOG_SIZE_IN_FIELDS {
+        /// Safety: randomness cannot be constrained.
+        final_log[i] = unsafe { random() };
+    }
+
```

Comment:
Would be nice to provide some commentary on wtf is going on in this and the previous for loops. Some concat fn would help readability as mentioned. But it's e.g. extremely nonobvious that we're adding padding here, and it's even more nonobvious _why_ we even want padding in the first place.

Reply:
I've added concat functions, and have explained why random padding is added.

Reply:
Unfortunately, I can't get a concat function to work to push the random padding to the log, because that requires the random padding size to be computed, and it seems the clever trick of doing the generics arithmetic in functions doesn't work for an expression so complex: It says it's overflowing some stack and kills the LSP.